[quant][graphmode] Test weight observer for dynamic quant

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39687 [quant][graphmode] Test weight observer for dynamic quant**

Summary:
Run the observer on the weight values and compare it with the calculated attributes in the graph

Test Plan:
python test/test_quantization.py test_dynamic_weight_observer

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21961907](https://our.internmc.facebook.com/intern/diff/D21961907)

supriyar committed:
[quant][graphmode] Test weight observer for dynamic quant
Summary:
Run the observer on the weight values and compare it with the calculated attributes in the graph

Test Plan:
python test/test_quantization.py test_dynamic_weight_observer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Test weight observer for dynamic quant"


Summary:
Run the observer on the weight values and compare it with the calculated attributes in the graph

Test Plan:
python test/test_quantization.py test_dynamic_weight_observer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2559,13 +2559,51 @@ def forward(self, x):
+        ref_qparams = []
+        qconfig = script_qconfig(default_dynamic_qconfig)
+        for wt in [model.res1.weight, model.res2.weight]:
+            get_forward(qconfig.weight)(wt)
+            qparams = qconfig.weight._get_method('calculate_qparams')()
```
I think you can call them directly, e.g. `qconfig.weight(wt)` and `qconfig.weight.calculate_qparams`

supriyar replied:
It doesn't seem to be directly callable `TypeError: 'torch._C.ScriptModule' object is not callable`

jerryzh168 replied:
I see, maybe you can wrap it into a RecursiveScriptModule with `wrap_cpp_module` first

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2559,13 +2559,51 @@ def forward(self, x):
+        qconfig = script_qconfig(default_dynamic_qconfig)
+        ref_qparams = []
+        for wt in [model.fc.weight, model.fc2.weight]:
+            get_forward(qconfig.weight)(wt)
+            qparams = qconfig.weight._get_method('calculate_qparams')()
```
same here

jerryzh168 reviewed (approved):
LG, see inline comment

supriyar committed:
Update on "[quant][graphmode] Test weight observer for dynamic quant"


Summary:
Run the observer on the weight values and compare it with the calculated attributes in the graph

Test Plan:
python test/test_quantization.py test_dynamic_weight_observer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
